src/segment_wifi.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type wifi struct {
+	props          *properties
+	env            environmentInfo
+	Connected      bool
+	State          string
+	SSID           string
+	RadioType      string
+	Authentication string
+	Channel        int
+	ReceiveRate    int
+	TransmitRate   int
+	Signal         int
+}
+
+const (
+	ConnectedIcon    Property = "connected_icon"
```

Comment:
These need to disappear in favor of a template that can handle swapping the icon with a simple if statement. We're trying to get rid of as muych icons as we can bny leveraging templates (you actually already have the logic in `defaultTemplate`)

Reply:
Do you mean, it'd be up to the user to use the template to prefix with their desired icon? And I should just bake this into the default template like `{{if .Connectected}}\uFAA8{{else}}\uFAA9{{end}}`?

Reply:
Exactly!

Reply:
Cool. This is much cleaner. Got rid of an oddity in the tests I didn't like too.